### ISSUE TITLE
Documentation is confusing: deploy with AZ CLI, acs-engine or....

### ISSUE THREAD

**Issue by tnolet** [Azure/acs-engine#1606]
This is a help request:

The documentation is very, very confusing with regard to what commands should be given with what tool to actually create a deployment. If I go to https://github.com/Azure/acs-engine/blob/master/docs/acsengine.md it tells me acs-engine generates templates, that you fill with relevant parameters. The output you than pass into the AZ CLI to actually deploy.

But following the Kubernetes walkthrough, https://github.com/Azure/acs-engine/blob/master/docs/kubernetes/deploy.md the situation is completely different. Now I need to pass in parameters to the acs-engine deploy command. Which one is recommended? Both?

**Comment by jackfrancis** [comment]
@tnolet Thanks for reporting. The project has evolved rapidly over the past year or so, and documentation reflects the aggregate knowledge of the various ways that you can deploy clusters, without effectively expressing strong opinions about the best ways to do things given certain conditions. We'll iron that out over time!

In the meantime, some clarification to your above comments. Both methods currently work. The reason you may want to just the "simple" `acs-engine deploy` method is to do a one-off cluster that may have little repeatable value. (Also, it is simpler overall, as it builds one command.) If you're building lots of clusters, however (especially lots of different flavors), the first method has advantages because you may want to iterate over bunch of different cluster declarations, and then run that through `acs-engine generate` to get the definitive template/params for sending to ARM. Also, those templates are re-usable and can be re-submitted many times using `az group deployment create ...` (caveats apply about resource collisions).

So, in other words, things have developed organically over time to be flexible, at the expense of user clarity. Please continue bugging us if you feel there's a better balance we can strike, and we especially take the feedback that our documentation would greatly benefit from a rationalization effort.

**Comment by tnolet** [comment]
@jackfrancis thanks, that helps!
